src/graphnet/training/labels.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+import numpy as np
+import torch
+from torch_geometric.data import Data
+from graphnet.utilities.logging import LoggerMixin
+
+
+class Label(ABC, LoggerMixin):
```

Comment:
I really like this, @RasmusOrsoe; I think it's quite elegant. We will have to consider whether we want _primarily_ to add these labels as part of the database production (e.g., trough similarly small `I3TruthExtractor` subclasses, like `I3TruthDirectionExtractor`), or primarily add them on-the-fly during training. But either way, I think it's nice to have both options, so I'd like to see this included. :)